http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -803,6 +813,38 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       )
     } yield ()
 
+  private def createRuntimeErrorHandler(msg: CreateRuntimeMessage, now: Instant)(e: Throwable): F[Unit] =
+    for {
+      _ <- logger.error(e)(s"Failed to create runtime ${msg.runtimeProjectAndName} in Google")
+      errorMessage = e match {
+        case leoEx: LeoException =>
+          Some(ErrorReport.loggableString(leoEx.toErrorReport))
+        case ee: com.google.api.gax.rpc.AbortedException
+            if ee.getStatusCode().getCode == 409 && ee.getMessage().contains("already exists") =>
+          None //this could happen when pubsub redelivers an event unexpectedly
+        case _ =>
+          Some(s"Failed to create cluster ${msg.runtimeProjectAndName} due to ${e.getMessage}")
+      }
+      _ <- errorMessage.traverse(m =>
+        (clusterErrorQuery.save(msg.runtimeId, RuntimeError(m, -1, now)) >>
+          clusterQuery.updateClusterStatus(msg.runtimeId, RuntimeStatus.Error, now)).transaction[F]
+      )
+      _ <- if (e.isReportWorthy)
+        errorReporting.reportError(e)
```

Comment:
Cool -- is `LeoPubsubMessageSubscriber` the one place where we do error reporting currently? I'd love to see a demo at some point (maybe you've given one already and I missed it..)

Reply:
yea, currently I'm only reporting the edge cases that I had left TODO previously (they're from monitor class, but will bubble up to subscriber and handled here)...

I gave a demo in today's walkthrough, but I can demo again during sprint review.